src/Microsoft.Data/PrimitiveColumnContainer.cs

```diff
@@ -188,55 +189,53 @@ public void ApplyElementwise(Func<T?, long, T?> func)
             {
                 ReadOnlyDataFrameBuffer<T> buffer = Buffers[b];
                 long prevLength = checked(Buffers[0].Length * b);
-                DataFrameBuffer<T> mutableBuffer = DataFrameBuffer<T>.GetMutableBuffer(buffer); 
-                Span<T> span = mutableBuffer.Span; 
-                for (int i = 0; i < buffer.Length; i++) 
+                DataFrameBuffer<T> mutableBuffer = DataFrameBuffer<T>.GetMutableBuffer(buffer);
+                Span<T> span = mutableBuffer.Span;
+                Span<byte> nullBitMapSpan = DataFrameBuffer<byte>.GetMutableBuffer(NullBitMapBuffers[b]).RawSpan;
+                for (int i = 0; i < span.Length; i++)
                 {
                     long curIndex = i + prevLength;
-                    this[curIndex] = func(IsValid(curIndex) ? span[i] : default(T?), curIndex);
-                } 
+                    bool isValid = IsValid(ref nullBitMapSpan, i);
+                    T? value = func(isValid ? span[i] : default(T?), curIndex);
+                    span[i] = value.GetValueOrDefault();
+                    SetValidityBit(nullBitMapSpan, i, value != null);
+                }
             }
         }
 
-        public bool IsValid(long index) => NullCount == 0 || GetValidityBit(index);
+        // Faster to use when we already have a span since it avoids indexing
+        public bool IsValid(ref Span<byte> bitMapBufferSpan, int index)
```

Comment:
You shouldn't need this overload - you should just need the one that takes a ReadOnlySpan.  Spans will implicitly convert to ReadOnlySpans.